state/application.go

```diff
@@ -2136,6 +2146,31 @@ func removeUnitResourcesOps(st *State, unitID string) ([]txn.Op, error) {
 	return ops, nil
 }
 
+func unassignUnitFromBranchOp(unitName, appName string, m *Model) (txn.Op, error) {
+	branch, err := m.unitBranch(unitName)
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+	if branch == nil {
+		return txn.Op{}, errors.NotFoundf("branch for %q", unitName)
```

Comment:
I would be inclined to return nil here and remove the specific detection from the caller - it's not really an exception case.

Reply:
done.